lemminx-maven/src/main/java/org/eclipse/lemminx/maven/PluginValidator.java

```diff
@@ -35,44 +35,41 @@ public PluginValidator(MavenProjectCache cache, RepositorySystemSession repoSess
 		this.repoSession = repoSession;
 		this.pluginManager = pluginManager;
 	}
-	
-	public boolean validatePluginResolution(DiagnosticRequest diagnosticRequest) {
+
+	public Diagnostic validatePluginResolution(DiagnosticRequest diagnosticRequest) {
```

Comment:
This could also be an `Optional<Diagnostic>`, or even an `Optional<List<Diagnostic>>` so this could be referenced in MavenDiagnosticParticipant.tagDiagnostics.

Reply:
+1